holmes/plugins/toolsets/prometheus/prometheus.py

```diff
@@ -245,23 +245,41 @@ def fetch_metadata_with_series_api(
     headers: Dict,
     config,
     verify_ssl: bool = True,
-) -> Dict:
+) -> Tuple[Dict, bool]:
+    """
+    Fetches metadata using the series API.
+
+    Returns:
+        Tuple[Dict, bool]: (metadata dict, is_truncated flag)
+    """
     url = urljoin(prometheus_url, "api/v1/series")
-    params: Dict = {"match[]": f'{{__name__=~".*{metric_name}.*"}}', "limit": "10000"}
+    limit = 100
+    # If pattern doesn't contain regex metacharacters, wrap with .* for backward compatibility
+    # Otherwise use as-is for full regex support
+    if not any(c in metric_name for c in r"^$.*+?{}[]|()\\"):
```

Comment:
Consider: extract the check to func and reuse or even better all the pattern logic in the if else.